code.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import xarray as xr
+
+lat = 43.07
+lon = 360-89.40
+
+# ERA5
+dset = xr.open_dataset('https://ds.nccs.nasa.gov/thredds/dodsC/bypass/CREATE-IP/reanalysis/ECMWF/IFS-Cy41r2/ERA5/mon/atmos/tas.ncml')
+era5 = dset['tas'].sel(lat=lat, lon=lon, method='nearest').compute() - 273.14
+
+# CMIP6
+#import pandas as pd
+#invdf = pd.read_csv('s3://cmip6-pds/pangeo-cmip6.csv')
+#use = invdf[
+#    (invdf['source_id'].isin(['CESM2','GFDL-ESM4','MIROC6'])) &
+#    (invdf['experiment_id'].isin(['historical','ssp585'])) &
+#    (invdf['member_id']=='r1i1p1f1') &
+#    (invdf['variable_id']=='pr') &
+#    (invdf['table_id']=='Amon')
+#]
+#use.to_csv('use_files.csv')
+#print(use)
+
+# MIROC6
+import s3fs
+fs = s3fs.S3FileSystem(anon=True)
+hist = xr.open_zarr(s3fs.S3Map('s3://cmip6-pds/CMIP6/CMIP/MIROC/MIROC6/historical/r1i1p1f1/Amon/tas/gn/v20181212', s3=fs), consolidated=True)['tas'] - 273.14
+future = xr.open_zarr(s3fs.S3Map('s3://cmip6-pds/CMIP6/ScenarioMIP/MIROC/MIROC6/ssp585/r1i1p1f1/Amon/tas/gn/v20190627', s3=fs), consolidated=True)['tas'] - 273.14
+miroc6 = xr.concat([hist,future], dim='time').sel(lat=lat, lon=lon, method='nearest').compute()
+
+# GFDL
+hist = xr.open_zarr(s3fs.S3Map('s3://cmip6-pds/CMIP6/CMIP/NOAA-GFDL/GFDL-ESM4/historical/r1i1p1f1/Amon/tas/gr1/v20190726', s3=fs), consolidated=True)['tas'] - 273.14
```

Comment:
Some of these blocks could be their own functions, with the order of subselecting optimized.